beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -52,6 +52,13 @@ func setExecutionData(ctx context.Context, blk interfaces.SignedBeaconBlock, loc
 	}
 
 	if localPayload == nil {
+		// Use builder payload if local payload is nil.
```

Comment:
By allowing either local payload or builder payload to be nil after these checks, then the switch statements below can have nil failures.

Reply:
I dont see how that is true.

The switch statement can only be entered if local and builders are not nil.

if both are nil: `errors.New("local and builder payloads are nil")`
if local nil, builder not nil: `blk.SetExecution(builderPayload)`
if local not nil, builder nil: `blk.SetExecution(localPayload)`
if both are not nil: enter switch

Reply:
this is not the case, we can have local==nil, builder != nil,  and then you set the execution payload but you still hit the switch statement.

Reply:
ah you are right, I thought I had it returned there

Reply:
9a5dc4f6c